x-pack/platform/plugins/shared/fleet/server/services/agents/status.ts

```diff
@@ -131,22 +132,25 @@ export async function getAgentStatusForAgentPolicy(
         esClient.search<
           null,
           { status: AggregationsTermsAggregateBase<AggregationsStatusTermsBucketKeys> }
-        >({
-          index: AGENTS_INDEX,
-          size: 0,
-          query,
-          fields: Object.keys(runtimeFields),
-          runtime_mappings: runtimeFields,
-          aggregations: {
-            status: {
-              terms: {
-                field: 'status',
-                size: Object.keys(statuses).length,
+        >(
+          {
+            index: AGENTS_INDEX,
+            size: 0,
+            query,
+            fields: Object.keys(runtimeFields),
+            runtime_mappings: runtimeFields,
+            aggregations: {
+              status: {
+                terms: {
+                  field: 'status',
+                  size: Object.keys(statuses).length,
+                },
               },
             },
+            ignore_unavailable: true,
```

Comment:
I think es query support `timeout` directly maybe we do not need an abort signal here